### PR TITLE
Deploy Istio and Gateway API

### PR DESCRIPTION
**Pull request by sergioifg94**
## Description

Add config and script steps to:
* Install the Gateway API in the control cluster and the workload clusters
* Install Istio in the workload clusters

The local-setup script now deploys these components by default. In order to verify this:

1. Run the local-setup with at least 1 workload cluster
    ```sh
    MCTC_WORKLOAD_CLUSTERS_COUNT=1 make local-setup
    ```
2. Once the setup finishes, verify that the control cluster has the Gateway API CRDs:
    ```sh
    kubectl get crds | grep gateway.networking
    gatewayclasses.gateway.networking.k8s.io   2023-02-01T14:47:46Z
    gateways.gateway.networking.k8s.io         2023-02-01T14:47:46Z
    httproutes.gateway.networking.k8s.io       2023-02-01T14:47:46Z
    ```
3. Perform the previous check on the workload cluster
    ```sh
   kubectl --context kind-mctc-workload-1 get crds | grep gateway.networking
    ```

### CODE REVIEW

File: hack/local-setup.sh [hunk 1]
```diff
@@ -25,10 +25,15 @@ KIND_CLUSTER_PREFIX="mctc-"
...
 
+BASE_PROXY_PORT=8085
+
 INGRESS_NGINX_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/ingress-nginx
 CERT_MANAGER_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/cert-manager
 EXTERNAL_DNS_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/external-dns
 ARGOCD_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/argocd
+ISTIO_KUSTOMIZATION_DIR=${LOCAL_SETUP_DIR}/../config/istio
+
+GATEWAY_API_RESOURCES_URL="github.com/kubernetes-sigs/gateway-api/config/crd?ref=v0.6.0"
```
Comment (maleck13) [inline hack/local-setup.sh:36]: do we want to pull this version out in to a var ?

**Comment by maleck13** [comment]
verfied setup worked, CRDS are present and Istio is running

**Review by maleck13 — APPROVED** [review]

File: config/istio/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+resources:
+  - namespace.yaml
+
+bases:
+  - ./base
+  - ./istiod
```
Comment (mikenairn) [inline config/istio/kustomization.yaml:6]: @sergioifg94 Why the 2 sets of resources here? They look to be the same?

Comment (mikenairn) [reply config/istio/kustomization.yaml:6]: I see now :eyeglasses: . Thanks

**Pull request merged**